engine/src/main/java/org/hibernate/search/engine/search/reference/FieldReference.java

```diff
@@ -0,0 +1,38 @@
+package org.hibernate.search.engine.search.reference;
+
+import org.hibernate.search.engine.search.common.ValueConvert;
+import org.hibernate.search.util.common.annotation.Incubating;
+
+/**
+ * @param <T> The expected returned type.
+ */
+@Incubating
+public interface FieldReference<T> {
+
+	String absolutePath();
+
+	default ValueConvert valueConvert() {
+		return ValueConvert.YES;
+	}
+
+	Class<T> type();
+
+	interface FieldAttributeReference<T, V> extends FieldReference<T> {
```

Comment:
I think you mean "a field that is also an attribute", which should IMO be named `AttributeFieldReference` rather than `FieldAttributeReference`... I understand `FieldAttributeReference` as "a reference to an attribute of a field" :/

But regardless... In the metamodel we have a `value`/`object` split rather than `attribute`/`object`... So I think this should be `ValueFieldReference`? `IndexValueFieldReference` maybe? `SearchValueFieldReference`?